packages/arb-token-bridge-ui/src/components/TransactionHistory/helpers.ts

```diff
@@ -445,16 +450,16 @@ export async function getUpdatedCctpTransfer(
       requiredL1BlocksBeforeConfirmation * blockTime
   if (
     // If transaction claim was set to failure, don't reset to Confirmed
-    tx.status !== 'Failure' &&
+    tx.status !== WithdrawalStatus.FAILURE &&
     isConfirmed
   ) {
     return {
       ...txWithTxId,
-      status: 'Confirmed'
+      status: WithdrawalStatus.CONFIRMED
     }
   }
 
-  return tx
+  return { ...tx, status: WithdrawalStatus.UNCONFIRMED }
```

Comment:
Since this is related code, we found an edge case yesterday where freshly claimed cctp deposits don't show `status` field in the tx history row, when moved from Pending to Settled tab.

The reason is that in case of cctp-withdrawal-claims, we correctly set the status to `EXECUTED`, but in case of cctp-deposit-claims, we set `status` as `undefined`, but `depositStatus` as `L2_SUCCESS`. 

When this updated tx goes to `useTransactionHistory` > `updatePendingTransaction`, it doesn't reach the cctp condition, as it returns earlier at `!isTxPending(tx)...`. Because of this, it's status remains undefined, and the row doesn't have any condition to handle that. It shows the status as null in JSX.

<img width="500" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/47498bf8-cae1-47f7-97a9-c725e637275e">


A fix for this can be, after claim, the status of the cctp deposit can also be EXECUTED, so that it shows the status badge, as shown here: 

<img width="500" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/d97eba47-a8ed-4ade-b235-b70bdd448e86">

Reply:
updated